76 featured paper

cosola opened the pull request:
#76 
initial commit

cosola committed:
completed framing features section

cosola committed:
featured-paper, first commit

cosola committed:
added button and included PDF

misscs commented on app/assets/styles/styles.css:
```diff
@@ -1,6 +1,9 @@
 
+* {
+  box-sizing: border-box;
+}
```
We should not need this. Did you make sure to base your branch on the most current work in `release-branch-v1`?

cosola replied:
Yes. It was not up-dated.

My intention was to understand why MR was having problem with this rule. 

It is now removed.

misscs commented on bulletin/index.html:
```diff
@@ -177,9 +177,55 @@ <h3 class="buarticle-scholarships--title">MACILE Scholarships Program </h3>
+			<div class="article">
+				<div class="article-banner">
+					<h1 class="article-banner--brand" itemprop="brand">MACILE</h1> 
+					<!-- <h3 class="article-banner--slogan">Advancing Engineering and Technology Education</h3> -->
```
The h1 headers are not correct HTML. You can't have an `h2` before and `h1`. `section` allows you to reset all headers. You could have an `h1` and then an `h2`.

Given that org name is just being used as a placeholder, you should not use an h1. It should be a span or some simple inline element because it's just being used as a presentation to create an image. Do you plan to replace this with you an image in the future? 

Also, you need to remove the `itemprop`. That breaks the schema. 

Please remove L188. All code that is commented out should always be removed.

cosola replied:
Why aren't h1 headers correct HTML?

I agree it is not correct to use h1 after h2. In my design plan this should be h2. No sure why I made the mistake.

True, h1 here is a placeholder for the MACILE logo; an image.

misscs replied:
My bad, that was not worded well. What I meant to say is that it isn't correct because it appears before the `h2`. Since it is a placeholder, can you use a `span`?

cosola replied:
Tried using span, but did not work well. Used h2 instead.

misscs replied:
You can't use an h2. it isn't semantic. Please see my additional commits. I fixed the issue.

misscs commented on bulletin/index.html:
```diff
@@ -177,9 +177,55 @@ <h3 class="buarticle-scholarships--title">MACILE Scholarships Program </h3>
+	</section>
+	<section class="feature-acknowledge">
+
+	</section>
```
What is this section? Why is it empty?

cosola replied:
No sure why is showing here. I removed it in my local repo of the branch.  
Could it be because the backup.html file is untracked?

misscs replied:
Why do you think it would have to do with an untracked file? Should an untrack file affect anything?

cosola replied:
Because it does not show in index.html in my local branch. It is in the file backup.html, which is not tracked.

misscs commented on app/assets/styles/bulletin.css:
```diff
@@ -176,6 +175,10 @@ hr.image {
 
+/* .article-content {
+	display: flex;
+} */
```
Please get in the habit of reviewing your PRs to make sure you remove simple errors, such as this. We don't allow code that is commented out into the repo.

cosola replied:
This is not an error.  I am trying to understand why the grid items in this grid are not behaving as expected when displayed flex.

misscs replied:
Okay. But the code has no value to the final PR. Please remove it.

cosola replied:
I found the answer and made the corresponding change.

misscs commented on app/assets/styles/bulletin.css:
```diff
@@ -194,9 +197,45 @@ hr.image {
+
+}
+
+a[href$=".pdf"]:hover::after {
+	color: white;
+}
```
This code should appear after L208 because it helps comprehension in the file. We group like blocks of code close together.

cosola replied:
I think it is properly placed because it follows the CSS code for  ".btn-small--yellow:hover { }".  The reason is that the rule "color: white" does not work when placed here. However, you are probably right.

Can you please explain, why I cannot change the font color within  ".btn-small--yellow:hover { }" ?

misscs replied:
Because `.btn-small--yellow` is styling a class and this is a global styling of all `a` elements. Therefore it should be grouped with the other ruleset that is styling the `a` element at the global level. 

Technically these should come at the top of the document where global elements are placed to leverage the cascade.

cosola replied:
I see.  But, why should this code be more properly placed after L208 instead of immediately after L222?

misscs commented on bulletin/index.html:
```diff
@@ -190,7 +190,9 @@ <h1 class="article-banner--brand" itemprop="brand">MACILE</h1>
-							MACILE is a research-based program founded to advance <span>STEM</span> education in less advantaged communities, engineering solutions to optimally increase access to robust<span>language-infused STEM</span> learning environments. Development follows three phases: exploration, design refinement, and expansion. We review results from 2007-2008. Clearly, focus on continuous improvement is p ...
+						MACILE is a research-based program founded to advance <span>STEM</span> education in less advantaged communities, engineering solutions to optimally increase access to robust<span>language-infused STEM</span> learning environments. This work review design improvements from 2007-2008.
+					</p>
+					<p ><a class="btn-small--yellow" href="/bulletin/macile-model-development-2007-08-ef.pdf"></a> 
```
HTML tags should not have extra spaces in the `<p>` not `<p >`.

You also added a PDF but the PDF was never added to the repo. I recommend putting the PDF at the root level in a folder called `content`.

cosola replied:
Done.

Did not realize the PDF file was untracked.

misscs commented on app/assets/styles/bulletin.css:
```diff
@@ -154,6 +153,89 @@ hr.image {
+	grid-gap: 1.5em;
+	grid-auto-flow: dense;
+	margin-top: 0;
+	/* align-items: center;
+	justify-items: center; */
```
Remove this code.

cosola replied:
Done

misscs reviewed (changes requested):
When you open a PR, you should take care to review the files to make sure you don't miss simple mistakes. It's like a final check on your work that often results in catching simple mistakes and errors.

misscs commented on app/assets/styles/bulletin.css:
```diff
@@ -154,6 +153,89 @@ hr.image {
+
+/* Including PDF ! */
+a[href$=".pdf"]:after { 
+	content: "Read more..."; 
```
I think this would be more compelling if it is just _Read more_. There's no need for the elipsis. Ellipsis is commonly used to mean something is truncated. Also, used to signify run off. This is neither of those cases. It is a call to action. Be affirmative.

cosola replied:
Done.

misscs commented on app/assets/styles/bulletin.css:
```diff
@@ -154,6 +153,89 @@ hr.image {
+/* Including PDF ! */
+a[href$=".pdf"]:after { 
+	content: "Read more..."; 
+	text-decoration: none;
```
You can remove L205. That rule is coming from the global stylesheet.

cosola replied:
Completed.

misscs commented on app/assets/styles/bulletin.css:
```diff
@@ -154,6 +153,89 @@ hr.image {
+a[href$=".pdf"]:after { 
+	content: "Read more..."; 
+	text-decoration: none;
+	font-size: 1.10em;
```
No need to add the 0. `1.1` means the same thing and is easier to read.

cosola replied:
I know it isn't necessary. Just an oversight.  Thanks for the alert.

cosola committed:
Included featured article; pdf file

cosola committed:
completed PR revisions

cosola committed:
removed empty css rule

cosola committed:
removed box-sizing:border-box from styles.css

cosola committed:
Removed ellipsis from btn; deleted text-decoration.

cosola committed:
Removed 0 from 1.10em

cosola committed:
Resolved comment on L226-L228.

misscs committed:
Change classes to featured-article

1. The correct term is featured article.
2. Added featured to classes that did not have it. We are following BEM, featured-article is the name of the block
3. Corrected BEM class structure

misscs committed:
Consolidate a class

The element had two classes applied that could be consolidated.

misscs committed:
Cleanup featured article image

Because this is an image we should not use a header element. It is not semantic HTML. I renamed this section to image because a banner is an element that typically spans the full width of a section and is used to highlight/announce information.

I added the aria role to signify to screen readers that this section is purely presentational, i.e. it doesn't have any content of value. It is just for visual presentation.

I removed the class `featurd-article__content` because it is not needed. Flex applies at the parent level.

In the future, I think adding an icon for PDF or paper would make it look much nicer.

misscs committed:
Remove class

This has no effect on the content

misscs committed:
Cleanup properties that don't have any effect

misscs committed:
Cleanup CSS and design

We can achieve a similar CSS effect with less code. There was also a lot of mixing of code and properties.

I aimemed to align the content horizontally with the grid above. Will need to adjust the article grid above later. It also is using a lot of extra code.

misscs committed:
Add display:block to button

Otherwise the <a> behaves as an inline element and the padding/margins don't cor
rectly apply.

misscs committed:
Remove dup properties + normalize padding to em

1. Changed padding to ems to keep consistent
2. Removed reduntant properties in the hover styles. Border radius and padding inherit from the parent.

misscs committed:
Group the media query

misscs committed:
Cleanup spacing

misscs committed:
Adjust fonts-size

The title and the section header had the same font-size. Visually they should be different. Also added line-height for readability.

misscs committed:
Increase line-height of abstract

1.4 or 1.3 is the common line-height used for text.

Removed a property that didn't have any effect.

misscs committed:
Cleanup featured article content

- adjust alignment
- add a max-width: this section needs a small max-width but we can handle that when we tweak the whole design a bit. the grid use more explicit widths so we don't have to declare the on the elements

misscs commented:
@cosola I pushed a bunch of commits. I cleaned up the CSS and fixed various semantic issues. Please take a look. This is ready for merge as is. We should make the final design adjustments when we take into consideration the full design.

misscs reviewed (approved):
I made changes that @cosola should review. It doesn't change the functionality. The design is slightly improved when it comes to legibility and hierarchy.

misscs committed:
Remove br tags

misscs committed:
Merge branch 'release-bulletin-v1' into 76-featured-paper
